src/Plugin/Field/FieldFormatter/StrawberryCitationFormatter.php

```diff
@@ -0,0 +1,588 @@
+   *   The Config factory.
+   * @param \Drupal\format_strawberryfield\EmbargoResolverInterface $embargo_resolver
+   *
+   * @param \Drupal\Core\File\FileSystemInterface $file_system
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, $label, $view_mode, array $third_party_settings, AccountInterface $current_user, EntityTypeManagerInterface $entity_type_manager, TwigEnvironment $twigEnvironment, ConfigFactoryInterface $config_factory, EmbargoResolverInterface $embargo_resolver, FileSystemInterface $file_system) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $label, $view_mode, $third_party_settings, $config_factory, $embargo_resolver, $current_user);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->twig = $twigEnvironment;
+    $this->fileSystem = $file_system;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['label'],
+      $configuration['view_mode'],
+      $configuration['third_party_settings'],
+      $container->get('current_user'),
+      $container->get('entity_type.manager'),
+      $container->get('twig'),
+      $container->get('config.factory'),
+      $container->get('format_strawberryfield.embargo_resolver'),
+      $container->get('file_system')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return parent::defaultSettings() + [
+        'label' => 'Descriptive Metadata',
+        'metadatadisplayentity_uuid' => NULL,
+        'metadatadisplayentity_uselabel' => TRUE,
+        'citationstyle' => NULL,
+      ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $entity = NULL;
+    if ($this->getSetting('metadatadisplayentity_uuid')) {
+      $entities = $this->entityTypeManager->getStorage('metadatadisplay_entity')->loadByProperties(['uuid' => $this->getSetting('metadatadisplayentity_uuid')]);
+      $entity = reset($entities);
+    }
+    //$citation_locales_file = '/var/www/html/vendor/citation-style-language/locales/locales.json';
+    //$citation_locales_file_contents = file_get_contents($citation_locales_file);
+    //$citation_locales = json_decode($citation_locales_file_contents, true);
+    //$locale_options = array();
+    //foreach($citation_locales['language-names'] as $key => $value) {
+    //  $locale_string = $value[1];
+    //  $locale_options[$key] = $this->t($locale_string);
+    //}
+    // There's a better way to get this directory
+    $citation_style_directory = '/var/www/html/vendor/citation-style-language/styles-distribution';
```

Comment:
`citation-style-language` library needs to be a composer dependency and yes, that is not a good way of getting that folder

Reply:
I will fix my own comment here. If we go composer we will need a custom script to git clone that repository. Or we can ship this as a DevOps operation and have a configuration (global one for archipelago) to state where this is downloaded/present

Reply:
Ok, this is already solved in the libraries example

{
    "name": "vendor-name/program-name",
    "repositories": [
        {
            "type": "package",
            "package": {
                "name": "citation-style-language/locales",
                "version":"1.0.0",
                "source": {
                    "type": "git",
                    "url": "https://github.com/citation-style-language/locales.git",
                    "reference": "master"
                }
            }
        },
        {
            "type": "package",
            "package": {
                "name": "citation-style-language/styles-distribution",
                "version":"1.0.0",
                "source": {
                    "type": "git",
                    "url": "https://github.com/citation-style-language/styles-distribution.git",
                    "reference": "master"
                }
            }
        }
    ],
    "require": {
        "citation-style-language/locales":"@dev",
        "citation-style-language/styles-distribution":"@dev",
        "seboettg/citeproc-php": "^2"
    }
}, so this needs to go in our composer, not just the "seboettg/citeproc-php": "^2"